### ISSUE TITLE
pulse amplitudes not show in schedule.draw()

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-terra#3485]
the amplitudes of pulses are not shown in a schedule and it's hard to know the scale.

![image](https://user-images.githubusercontent.com/8622381/69014884-5408ff00-0954-11ea-9c37-5cee20f6509a.png)

single pulse visualizations (commands) have this information
![image](https://user-images.githubusercontent.com/8622381/69014929-93cfe680-0954-11ea-956a-415c62a4f7be.png)

**Comment by pushpulldecoder** [comment]
Is this issue still open?

**Comment by taalexander** [comment]
Hello @pushpulldecoder, this is still open. Do you wish me to assign this issue to you? If so I believe we should make this an option `show_yaxis`.

**Comment by pushpulldecoder** [comment]
Sure I would like to work on it.

**Comment by taalexander** [comment]
Assigned 😄

**Comment by pushpulldecoder** [comment]
![image](https://user-images.githubusercontent.com/43278046/89403341-c49bba00-d735-11ea-90c1-a7f347aa40e2.png)

So, I have to make this label within red optional?

**Comment by coder-mtv** [comment]
Hey! Is anybody still working on it?

**Comment by taalexander** [comment]
@pushpulldecoder, sorry about the delay in my response. I believe we need to add amplitude values only the y-axis like @ajavadia's second image shows

@coder-mtv - This is currently assigned to @pushpulldecoder. If for some reason they cannot undertake the issue I can reassign to you.

**Comment by taalexander** [comment]
@coder-mtv, it seems little progress has been made by @pushpulldecoder. I can reassign to you if you would like?

**Comment by coder-mtv** [comment]
That would be great! @taalexander

**Comment by taalexander** [comment]
Awesome, assigned!

**Comment by vishalg2235** [comment]
Hi @coder-mtv are you still working on this? I want to give it a try!

I have tried to gain some knowledge from online resource regarding Pulse Programming Model, Pulse Schedule and Instructions. I don't think I understood them completely but if someone can tell me how(code) to replicate this issue. Then I think I can further use my Python knowledge to debug it.
@taalexander @ajavadia

**Comment by taalexander** [comment]
Hello @vishalg2235, I have assigned.

Something like:

```python
from qiskit import pulse

waveform = pulse.Waveform([0.0, 0.1, 0.2, 0.3, 0.4])
waveform.draw()
```

Should allow you to generate the plot with an axis and for without an axis add:

```python
sched = pulse.Schedule()
sched += pulse.Play(waveform, pulse.DriveChannel(0))
sched.draw()
```

**Comment by vishalg2235** [comment]
Thanks @taalexander

**Comment by vishalg2235** [comment]
Hi @taalexander I found the line of code responsible for the issue [here](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/visualization/pulse/matplotlib.py#L857)
After commenting out that line I found following plot. Have a look at the yellow circled zone where labels overlap each other.

![Screenshot 2020-11-25 at 10 17 34 AM](https://user-images.githubusercontent.com/17894780/100184452-8b1ad480-2f07-11eb-9699-76b59b147662.png)

**Comment by vishalg2235** [comment]
When I removed X-axis limit then it somewhat adjusts itself to display without overlap.
![Screenshot 2020-11-25 at 12 19 35 PM](https://user-images.githubusercontent.com/17894780/100192858-b5c15900-2f18-11eb-89b6-cde94cfe2ad8.png)

**Comment by vishalg2235** [comment]
@taalexander in multiple channels the Y-axis labels can be misleading/wrong as you can see in pic below that for 'd0' channel labeled amplitude is correct but for channel 'd1' its wrong:

![Screenshot 2020-11-26 at 12 42 33 AM](https://user-images.githubusercontent.com/17894780/100272135-a118ab80-2f80-11eb-835a-d5add4f58e8c.png)

**Comment by taalexander** [comment]
I believe the values for the y-axis would need to be manually added. I see the problem now, @nkanazawa1989 may know how to better resolve this.

**Comment by vishalg2235** [comment]
Ok meanwhile I'm also getting an idea of filling Yaxis values based on number of pulses and make a list of them and finally using that list in the graph. I will show it here once done with it.

**Comment by taalexander** [comment]
Any update here @nkanazawa1989?

**Comment by nkanazawa1989** [comment]
I don't think we updated this. In v2 still yaxis value is missing, but with debugging stylesheet we can show max/min amplitude for each pulse.

**Comment by javabster** [comment]
Hi @vishalg2235 are you still working on this? Let us know if you have any further questions 😄

**Comment by vishalg2235** [comment]
Yes @javabster I was hoping for a better explanation. I will look deeply into this again and comment here all my doubts shortly.

**Comment by ajavadia** [comment]
@nkanazawa1989 can you paste an example picture from the v2 debugging stylesheet? maybe we just go with that

**Comment by mrvee-qC** [comment]
# Update for Santa- QGoGP
✅ Issue reproducible! Y-axis ticks are not implemented v2 version of pulse visualization and older issues of v1 still exist as discussed above!

Will update comment/add more comments if cause is found/fixed!

> Python version 3.9.7
> qiskit-terra version: 0.19.1

**Comment by 1ucian0** [comment]
ping @vishalg2235 ?

**Comment by HuangJunye** [comment]
@vishalg2235 Hey, just want to check in to see if you are still working on this issue. Please let us know if you need any help!

**Comment by javabster** [comment]
as there is no response from the original assignee I am un-assigning so this issue is now open for others to pick up 😄

**Comment by A-Tessier01** [comment]
Hi, I am looking for a first issue to work on, could this be assigned to me if no one is currently working on it?

**Comment by javabster** [comment]
Thanks for volunteering @A-Tessier01, I'll assing to you :) Please make sure to read the c[ontributing guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md), and take a look at [this youtube series](https://www.youtube.com/watch?v=mETgusQICOA&list=PLOFEBzvs-VvqKza-VLSYzWxnUMp-nO_aS&index=6) for github support

**Comment by shravanpatel30** [comment]
Hi, Is this issue still open? I'm looking for a good first issue to contribute. 

And if this issue is still available, then we want to get amplitude like below image, right?
 
![image](https://user-images.githubusercontent.com/39517270/137504688-4ca1c825-11c1-40b7-8588-a01cf2d0c784.png)

**Comment by taalexander** [comment]
@nkanazawa1989 do you know if this issue is still open?

**Comment by 1ucian0** [comment]
`qiskit.pulse` was deprecated in 1.3 and removed in 2.0. It has not been supported since 1.4 reached end-of-life in April 2026. Closing this issue as `not planned`.
